### PR TITLE
Write DiagnosticSuppressor for CS8618

### LINKED ISSUE #21608: Write a DiagnosticSuppressor for CS8618 for DbSet properties
See https://github.com/dotnet/roslyn/issues/45912

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #21608

This is not ready yet. I need some guidance on writing the unit test, as well as reviewing the suppressor and guidance on the TODO comment.

cc: @ajcvickers @roji

### CODE REVIEW

File: test/EFCore.Analyzers.Tests/NullabilityDiagnosticSuppressorTest.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using EFCore.Analyzers.Test.TestUtilities;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Xunit;
+
+namespace Microsoft.EntityFrameworkCore
+{
+    public class NullabilityDiagnosticSuppressorTest
```
Comment (roji) [inline test/EFCore.Analyzers.Tests/NullabilityDiagnosticSuppressorTest.cs:12]: I recommend you take a look at InternalUsageDiagnosticAnalyzerTest and try to do something similar.

File: src/EFCore.Analyzers/NullabilityDiagnosticSuppressor.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                if (!(root.FindNode(diagnostic.Location.SourceSpan) is PropertyDeclarationSyntax node))
+                {
+                    continue;
+                }
+                if (!(node.Parent is ClassDeclarationSyntax classDeclaration))
+                {
+                    continue;
+                }
+
+                // TODO: Check if the class derives from DbContext. (Not sure how to do it yet)
```
Comment (roji) [inline src/EFCore.Analyzers/NullabilityDiagnosticSuppressor.cs:32]: The code as is only looking at syntax nodes, which isn't sufficient for identifying inheritance AFAIK - you need to look at the semantic model (GetSymbolInfo). However, GetSymbolInfo apparently has some perf issues, so #18637 is about modifying our current analyzer to use IOperation instead; am not sure exactly how that looks like for a suppressor but it should be possible.

At least for now, our analyzers package does cannot reference EF Core types directly (I'll try to see about improving this in the future), so you can only identify/recognize EF types by name.

File: test/EFCore.Analyzers.Tests/NullabilityDiagnosticSuppressorTest.cs (hunk 1 above)
Comment (Youssef1313) [reply test/EFCore.Analyzers.Tests/NullabilityDiagnosticSuppressorTest.cs:12]: @roji I tried to do that. But seems like the current BaseTest won't fit well with testing suppressor.

File: src/EFCore.Analyzers/NullabilityDiagnosticSuppressor.cs (hunk 2 above)
Comment (Youssef1313) [reply src/EFCore.Analyzers/NullabilityDiagnosticSuppressor.cs:32]: @roji, I'll try to play around with SemanticModel.

File: test/EFCore.Analyzers.Tests/NullabilityDiagnosticSuppressorTest.cs (hunk 1 above)
Comment (roji) [reply test/EFCore.Analyzers.Tests/NullabilityDiagnosticSuppressorTest.cs:12]: In that case I'd look at some other well-known suppressor and try to see what it does..

File: src/EFCore.Analyzers/NullabilityDiagnosticSuppressor.cs (hunk 2 above)
Comment (roji) [reply src/EFCore.Analyzers/NullabilityDiagnosticSuppressor.cs:32]: @Youssef1313 please hold off on this, @davidroth's suggestion in #21608 will likely obviate this.

File: test/EFCore.Analyzers.Tests/NullabilityDiagnosticSuppressorTest.cs (hunk 1 above)
Comment (sharwell) [reply test/EFCore.Analyzers.Tests/NullabilityDiagnosticSuppressorTest.cs:12]: We don't have a good way to test suppressors currently.

**Comment by roji** [comment]
@Youssef1313 am closing this as @davidroth's suggestion to simply initialize the DbSet properties is cleaner and obviates this - but thanks for your work on this.

There's still #21663 which could definitely benefit from a suppressor, but please note it's unlikely we'll merge this for 5.0 this late.

**Pull request closed without merging**